pytket/pytket/passes/auto_rebase.py

```diff
@@ -85,21 +107,29 @@ def auto_rebase_pass(gateset: Set[OpType]) -> RebaseCustom:
     """Attempt to generate a rebase pass automatically for the given target
     gateset.
 
-    Checks if there are known existing decompositions from CX
+    Checks if there are known existing decompositions
     to target gateset and TK1 to target gateset and uses those to construct a
     custom rebase.
     Raises an error if no known decompositions can be found, in which case try
     using RebaseCustom with your own decompositions.
 
     :param gateset: Set of supported OpTypes, target gate set.
     :type gateset: FrozenSet[OpType]
-    :raises NoAutoRebase: No suitable CX or TK1 decomposition found.
+    :raises NoAutoRebase: No suitable decomposition found.
     :return: Rebase pass.
     :rtype: RebaseCustom
     """
-    return RebaseCustom(
-        gateset, get_cx_decomposition(gateset), get_TK1_decomposition_function(gateset)
-    )
+    tk1 = get_TK1_decomposition_function(gateset)
+    try:
+        return RebaseCustom(gateset, tk1, get_tk2_decomposition(gateset))
+    except NoAutoRebase:
+        pass
+    try:
+        return RebaseCustom(gateset, get_cx_decomposition(gateset), tk1)
```

Comment:
The order of the parameters here is different. Is this a bug or an inconsistent API?

Reply:
It's an inconsistent API. I thought it more natural to have TK1 before TK2 but perhaps it is better to keep consistency and have the TK1 replacement function last.

Reply:
Swapped/

Reply:
I agree with you, but I think the change is worth it for the sake of consistency.